### PR TITLE
Simplify and clean CMake configuration

### PR DESCRIPTION
**Pull request by JavierCVilla**
- Remove unneeded `FindROOT.cmake` file: ROOT can be found using `CMAKE_PREFIX_PATH`
- Remove quotes from directories specified in `include_directories`
- Link only a subset of root libraries, only those strictly needed
- Remove redundant linked libraries

### CODE REVIEW

File: utilities/CMakeLists.txt [hunk 1]
```diff
@@ -6,7 +6,7 @@ file(GLOB sources *.cc)
 file(GLOB headers *.h)
 
 add_library(utilities SHARED ${sources} ${headers} )
-target_link_libraries(utilities datamodel podio ${ROOT_LIBRARIES} ${ROOT_COMPONENT_LIBRARIES})
+target_link_libraries(utilities datamodel podio ROOT::Physics ${ROOT_COMPONENT_LIBRARIES})
```
Comment (vvolkl) [inline utilities/CMakeLists.txt]: What is `${ROOT_COMPONENT_LIBRARIES}` here actually? Can we remove it and add Root targets also?

**Review by vvolkl — CHANGES_REQUESTED** [review]
Very nice, thanks Javier. Could you squash the last two commits?

File: utilities/CMakeLists.txt (hunk 1 above)
Comment (JavierCVilla) [reply utilities/CMakeLists.txt]: It's indeed empty, so I removed it.

**Comment by JavierCVilla** [comment]
@phsft-bot build

**Comment by JavierCVilla** [comment]
@vvolkl thanks for the review, all comments addressed

**Review by vvolkl — APPROVED** [review]

**Pull request merged**